app/assets/javascripts/short-form/ShortFormApplicationController.js.coffee

```diff
@@ -266,6 +266,8 @@ ShortFormApplicationController = (
   $scope.validateHouseholdEligibility = (match, callbackUrl) ->
     $scope.clearHouseholdErrorMessage()
     form = $scope.form.applicationForm
+    if match == 'incomeMatch' && $scope.applicant.household_vouchers_subsidies == 'Yes'
+      $state.go(callbackUrl)
     ShortFormApplicationService.checkHouseholdEligiblity($scope.listing)
```

Comment:
^^ this is really the only change needed for this feature, however because of some changes on the Salesforce side I also ended up regenerating some of the VCR cassettes/fixtures and did some other minor refactoring / typo-fixing so that also went into this PR